java/src/test/java/ai/rapids/cudf/TableTest.java

```diff
@@ -6080,6 +6077,167 @@ void testAllFilteredFromValidity() {
+    assert (validity.length >= BitVectorHelper.getValidityAllocationSizeInBytes(cv.rows));
+    if (cv.type.isNestedType()) {
+      ColumnView[] children = cv.getChildColumnViews();
+      try {
+        return new ColumnView(cv.type,
+            cv.rows,
+            Optional.of(0L),
+            validity,
+            cv.getOffsets(),
+            children);
+      } finally {
+        for (ColumnView v : children) {
+          if (v != null) {
+            v.close();
+          }
+        }
+      }
+    } else {
+      return new ColumnView(cv.type, cv.rows, Optional.of(0L), cv.getData(), validity, cv.getOffsets());
+    }
+  }
+
+  @Test
+  void testRemoveNullMasksIfNeededNoNulls() {
+    ListType nestedType = new ListType(true, new StructType(false,
+        new BasicType(true, DType.INT32),
+        new BasicType(true, DType.INT64)));
+
+    List data1 = Arrays.asList(10, 20L);
+    List data2 = Arrays.asList(50, 60L);
+    HostColumnVector.StructData structData1 = new HostColumnVector.StructData(data1);
+    HostColumnVector.StructData structData2 = new HostColumnVector.StructData(data2);
+
+    //First we create ColumnVectors
+    try (ColumnVector nonNullVector0 = ColumnVector.fromBoxedInts(1, 2, 3);
+         ColumnVector nonNullVector2 = ColumnVector.fromStrings("1", "2", "3");
+         ColumnVector nonNullVector1 = ColumnVector.fromLists(nestedType,
+             Arrays.asList(structData1, structData2),
+             Arrays.asList(structData1, structData2),
+             Arrays.asList(structData1, structData2))) {
+      //Then we take the created ColumnVectors and add validity masks even though the nullCount = 0
+      long allocSize = BitVectorHelper.getValidityAllocationSizeInBytes(nonNullVector0.rows);
+      DeviceMemoryBuffer dm0 = DeviceMemoryBuffer.allocate(allocSize);
+      Cuda.memset(dm0.address, (byte) 0xFF, allocSize);
+      DeviceMemoryBuffer dm1 = DeviceMemoryBuffer.allocate(allocSize);
+      Cuda.memset(dm1.address, (byte) 0xFF, allocSize);
+      DeviceMemoryBuffer dm2 = DeviceMemoryBuffer.allocate(allocSize);
+      Cuda.memset(dm2.address, (byte) 0xFF, allocSize);
+      DeviceMemoryBuffer dm3_child = DeviceMemoryBuffer.allocate(BitVectorHelper.getValidityAllocationSizeInBytes(2));
+      Cuda.memset(dm3_child.address, (byte) 0xFF, BitVectorHelper.getValidityAllocationSizeInBytes(2));
+
+      try (ColumnVector cv0 = replaceValidity(nonNullVector0, dm0).copyToColumnVector();
+           ColumnView struct = nonNullVector1.getChildColumnView(0);
+           ColumnView structChild0 = struct.getChildColumnView(0);
+           ColumnView newStructChild0 = replaceValidity(structChild0, dm3_child);
+           ColumnView newStruct = struct.replaceChildrenWithViews(new int[]{0}, new ColumnView[]{newStructChild0});
+           ColumnView list = nonNullVector1.replaceChildrenWithViews(new int[]{0}, new ColumnView[]{newStruct});
+           ColumnVector cv1 = replaceValidity(list, dm1).copyToColumnVector();
```

Comment:
The view from the replace validity is leaked.

Reply:
I dislike the name `getChildColumnView` considering that we have a `getColumn` which simply returns a reference to an existing vector and no cleanup is necessary. I wish we could name it something like `createChildColumnView` so the user knows that a new `ColumnView` is being returned and needs to be cleaned. But tbh I don't know if I am the only one who seems to miss cleaning up the ColumnViews or if there are others.